validator/sawtooth_validator/journal/consensus/dev_mode/dev_mode_consensus.py

```diff
@@ -81,23 +144,44 @@ def __init__(self, block_cache, state_view):
     def verify_block(self, block_state):
         return True
 
+    def compute_block_weight(self, block_state):
+        # longest chain wins
+        return block_state.block_num
+
 
 class ForkResolver(ForkResolverInterface):
-    """ DevMode ForkResolver implementation
-    """
+    # Provides the fork resolution interface for the BlockValidator to use
+    # when deciding between 2 forks.
     def __init__(self, block_cache):
         super().__init__(block_cache)
 
         self._block_cache = block_cache
 
     def compare_forks(self, cur_fork_head, new_fork_head):
         """
+
         Args:
-            cur_fork_head (BlockWrapper): The current head of the block chain.
-            new_fork_head (BlockWrapper): The head of the fork that is being
-            evaluated.
+            cur_fork_head: The current head of the block chain.
+            new_fork_head: The head of the fork that is being evaluated.
         Returns:
-            bool: True if the new chain should replace the current chain.
-            False if the new chain should be discarded.
+            bool: True if the result of the signer_pubkey hashed with the
+             header_signature of the new fork head is the lowest value.
         """
-        return new_fork_head.block_num > cur_fork_head.block_num
+
+        cur_fork_hash = hash_signer_pubkey(
+            self.cur_fork_head.block_header.signer_pubkey,
```

Comment:
These are not class members. 
access as cur_fork_head.block_header.signer_pubkey not self.cur_fork_head.block_header.signer_pubkey